mri/reconstructors/tests/test_reconstructors.py

```diff
@@ -0,0 +1,271 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+# Third-party import
+import numpy as np
+import unittest
+
+# Package import
+from mri.operators.fourier.cartesian import FFT
+from mri.operators.fourier.non_cartesian import NonCartesianFFT, Stacked3DNFFT
+from mri.operators.linear.wavelet import WaveletUD2, WaveletN
+from mri.reconstructors import SingleChannelReconstructor, \
+    SelfCalibrationReconstructor, CalibrationlessReconstructor
+from mri.operators.utils import convert_mask_to_locations
+from pysap.data import get_sample_data
+
+from itertools import product
+from modopt.opt.proximity import SparseThreshold
+from modopt.opt.linear import Identity
+
+
+class TestReconstructor(unittest.TestCase):
+    """ Tests every reconstructor with mu=0, a value to which we know the
+    solution must converge to analytical solution,
+    ie the inverse fourier transform
+    """
+    def setUp(self):
+        """ Setup common variables to be used in tests:
+        num_iter : Number of iterations
+        images : Ground truth images to test with, obtained from server
```

Comment:
Obatining the data from the server might lengthen the tests, I guess that at some point the best thing to do is to generate random complex data

Reply:
This will be another PR. Writing TODO